frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+  Typography,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { post } from '../../lib/k8s/apiProxy';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, AuthVisible } from '../common';
+
+export function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+
+  const dispatch = useDispatch();
+  const kubeNamespace = Namespace;
+
+  function toggleCreateNamespaceVisibility() {
+    setNamespaceDialogOpen(namespaceDialogOpen => !namespaceDialogOpen);
+  }
+
+  console.log('kubeNamespace', kubeNamespace);
+
+  function createNewNamespace() {
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+      },
+    };
+
+    // need this for the item name?
+    const itemName = newNamespaceData.metadata.name;
+
+    // need a callback so just made one to use
+    const callback = (itemName: string) => {
+      console.log(`Callback called with ${itemName}`);
+    };
+
+    async function createNamesapce(newNamespaceData: any) {
+      try {
+        const response = await post('/api/v1/namespaces', newNamespaceData);
```

Comment:
We need to pass the cluster here. And this function can only be called after the grace period. i.e. it should be passed to the `clusterAction`.

Reply:
post logic currently handles setting the cluster

Reply:
Here is the problem: the post handles setting the cluster, but post is called only after the grace period, and by then the user may have changed to another cluster (and that's the one the post picks up). This is why we had to add the cluster as an arg to these functions. In the future we should have the cluster as a mandatory argument to avoid this.